### PR TITLE
Small refactoring to load from ckpt

### PR DESCRIPTION
**Pull request by sfmig**
Following suggestions from a similar code review, this PR
- moves the computation of the checkpoint type to `detection_utils`,
- refactors the main training script to reduce length.

Let me know what you think!

thanks 🙌

### CODE REVIEW

File: crabs/detection_tracking/train_model.py [hunk 1]
```diff
@@ -164,37 +165,15 @@ def core_training(self) -> lightning.Trainer:
...
-                )
-            else:
-                checkpoint_type = "weights"  # for fine tuning
-                logging.info(
-                    f"Fine-tuning training from checkpoint at: {self.checkpoint_path}"
-                )
-        else:
-            checkpoint_type = None
+        # Get checkpoint type: "full", "weights" or None
+        checkpoint_type = get_checkpoint_type(self.checkpoint_path)
```
Comment (nikk-nikaznan) [inline crabs/detection_tracking/train_model.py:169]:
```suggestion
if self.checkpoint_path:
        checkpoint_type = get_checkpoint_type(self.checkpoint_path)
 else:
        checkpoint_type = None
```

**Review by nikk-nikaznan — APPROVED** [review]
Yes, make sense to me. Just small suggestion, should we only call the function if there is checkpoint_path? So we also don't need to check the condition, if the path is not there

File: crabs/detection_tracking/train_model.py (hunk 1 above)
Comment (sfmig) [reply crabs/detection_tracking/train_model.py]: yes fantastic! 🙌

**Review by nikk-nikaznan — APPROVED** [review]

**Comment by sfmig** [comment]
By mistake I merged PR #141 before this one, instead of merging this --> rebasing PR #141 ---> merging PR #141.

Just leaving a log here in case we need to inspect the commit history at some point and are confused about it.

**Pull request merged**